vegeta_wrapper/trigger_vegeta.py

```diff
@@ -31,20 +31,39 @@ def __init__(self, args):
         self.duration = args.duration
         self.cluster_name = args.cluster_name
         self.keepalive = args.keepalive
+        if args.results :
+            self.target_name = args.target_name
+            self.results = args.results
+        else :
+            self.results = None
 
     def _json_payload(self, data, sample):
-        payload = {
-            "workload": "vegeta",
-            "uuid": self.uuid,
-            "user": self.user,
-            "cluster_name": self.cluster_name,
-            "iteration": sample,
-            "duration": self.duration,
-            "workers": self.workers,
-            "keepalive": self.keepalive,
-            "targets": os.path.basename(self.targets),
-            "hostname": socket.gethostname()
-        }
+        if self.results:
+            payload = {
+                "workload": "vegeta",
+                "uuid": self.uuid,
+                "user": self.user,
+                "cluster_name": self.cluster_name,
+                "iteration": sample,
+                "duration": self.duration,
+                "workers": self.workers,
+                "keepalive": self.keepalive,
+                "targets": self.target_name,
+                "hostname": socket.gethostname()
+            }
+        else :
+            payload = {
+                "workload": "vegeta",
+                "uuid": self.uuid,
+                "user": self.user,
+                "cluster_name": self.cluster_name,
+                "iteration": sample,
+                "duration": self.duration,
+                "workers": self.workers,
+                "keepalive": self.keepalive,
+                "targets": os.path.basename(self.targets),
+                "hostname": socket.gethostname()
+            }
```

Comment:
This could be simplified with something like:

```suggestion
        if self.results:
            targets = serf.target_name 
        else:
            targets = os.path.basename(self.targets)
        payload = {
            "workload": "vegeta",
            "uuid": self.uuid,
            "user": self.user,
            "cluster_name": self.cluster_name,
            "iteration": sample,
            "duration": self.duration,
            "workers": self.workers,
            "keepalive": self.keepalive,
            "targets": targets,
            "hostname": socket.gethostname()
        }
```

Reply:
:+1: Thanks!